rfc822/parser.go

```diff
@@ -155,6 +155,10 @@ func Split(b []byte) ([]byte, []byte) {
 func parse(literal []byte, identifier []int, begin, end int) *Section {
 	header, _ := Split(literal[begin:end])
 
+	if !isHeader(header) {
```

Comment:
Can we update this piece of code so that when we have a valid header you cache the result in the section for the next time we re-use the parsed header here?

Reply:
done